### PR TITLE
Rename gesv to solve

### PR DESCRIPTION
**Pull request by vishwakftw**
Changelog:

- Renames `gesv` to `solve` to remain consistent with `cholesky_solve`.
- Rename all tests, fix callsites
- Create a tentative alias for `solve` under the name `gesv`, and add a deprecated warning to not promote usage.

Test Plan:

- All tests should pass to confirm the patch is correct

### CODE REVIEW

File: aten/src/ATen/core/TensorMethods.h [hunk 1]
```diff
@@ -1159,6 +1156,9 @@ inline Tensor Tensor::cholesky(bool upper) const {
 inline Tensor Tensor::cholesky_solve(const Tensor & input2, bool upper) const {
     return type().cholesky_solve(*this, input2, upper);
 }
+inline std::tuple<Tensor,Tensor> Tensor::solve(const Tensor & A) const {
```
Comment (zou3519) [inline aten/src/ATen/core/TensorMethods.h]: Is this file autogenerated?

File: tools/autograd/gen_python_functions.py [hunk 2]
```diff
@@ -26,8 +26,8 @@
     '_indexCopy_', 'max_values', 'min_values', 'argmax', 'argmin',
     '_cumsum.*', '_cumprod.*', '_sum.*', '_prod.*',
     '_th_.*', '_thnn_.*',
-    'arange.*', 'range.*', '_gesv.*', '_getri.*', '_inverse.*',
-    '_potrs.*', '_cholesky.*', '_btrifact.*',
+    'arange.*', 'range.*', '_solve.*', '_getri.*', '_inverse.*',
+    '_cholesky.*', '_btrifact.*',
```
Comment (zou3519) [inline tools/autograd/gen_python_functions.py]: What happened to `'_potrs.*',` ?

**Review by zou3519 — APPROVED** [review]

File: aten/src/ATen/core/TensorMethods.h (hunk 1 above)
Comment (vishwakftw) [reply aten/src/ATen/core/TensorMethods.h]: This file is auto-generated separately, and then compared against this file. If this is not modified, then an error is thrown during the build process.

File: tools/autograd/gen_python_functions.py (hunk 2 above)
Comment (vishwakftw) [reply tools/autograd/gen_python_functions.py]: I forgot to remove this while renaming `potrs` to `cholesky_solve`, so this should not cause any issues whatsoever.

**Comment by vishwakftw** [comment]
Thank you for the review, @zou3519.

**Comment by vishwakftw** [comment]
Should I go ahead and ask pytorchbot to merge this in?

**Comment by zou3519** [comment]
@vishwakftw nope, I got it :)

**Pull request closed without merging**